### PR TITLE
fix(eslint): manual fix of eslint errors on core

### PR DESCRIPTION
**Pull request by matthieu-crouzet**

### CODE REVIEW

File: packages/@o3r/core/src/store/types.ts [hunk 1]
```diff
@@ -65,27 +65,32 @@ export interface FailActionPayload<T> {
   error?: T;
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export type keep<T, K extends keyof T> = (Partial<T> & { [p in K]: T[p] });
+export type Keep<T, K extends keyof T> = (Partial<T> & { [p in K]: T[p] });
```
Comment (fpaul-1A) [inline packages/@o3r/core/src/store/types.ts:68]: isn't it basically just a `Pick` ?

Comment (matthieu-crouzet) [reply packages/@o3r/core/src/store/types.ts:68]: I don't want to refactor the code in this PR
I think Pick is equivalent to `type Pick = { [p in K]: T[p] }`
so here we keep all the fields but some becomes non mandatory but if you defined them you still have to match the initial typing

Comment (fpaul-1A) [reply packages/@o3r/core/src/store/types.ts:68]: yes I read too fast
```suggestion
export type Keep<T, K extends keyof T> = Partial<T> & Pick<T,K>;
```

**Review by fpaul-1A — APPROVED** [review]

File: packages/@o3r/core/src/store/types.ts [hunk 2]
```diff
@@ -65,27 +65,32 @@ export interface FailActionPayload<T> {
   error?: T;
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export type keep<T, K extends keyof T> = (Partial<T> & { [p in K]: T[p] });
+export type Keep<T, K extends keyof T> = Partial<T> & Pick<T, K>;
+
+/**
+ * @deprecated please use Keep instead, will be removed in v13.
```
Comment (kpanot) [inline packages/@o3r/core/src/store/types.ts:71]:
```suggestion
 * @deprecated please use {@link Keep} instead, will be removed in v13.
```

**Review by kpanot — APPROVED** [review]

**Pull request merged**